moped-editor/src/queries/components.js

```diff
@@ -10,10 +10,93 @@ export const GET_COMPONENTS_FORM_OPTIONS = gql`
       component_name
       component_subtype
       line_representation
+      feature_layer {
+        internal_table
+      }
       moped_subcomponents(order_by: { subcomponent_name: asc }) {
         subcomponent_id
         subcomponent_name
       }
     }
   }
 `;
+
+export const ADD_PROJECT_COMPONENT = gql`
+  mutation AddProjectComponent($object: moped_proj_components_insert_input!) {
+    insert_moped_proj_components_one(object: $object) {
+      component_id
+    }
+  }
+`;
+
+export const GET_PROJECT_COMPONENTS = gql`
+  query GetProjectComponents($projectId: Int!) {
+    moped_proj_components(
+      where: { project_id: { _eq: $projectId }, is_deleted: { _eq: false } }
+    ) {
+      project_component_id
+      component_id
+      description
+      moped_components {
+        component_name
+        component_subtype
+        feature_layer {
+          internal_table
+        }
+        line_representation
+      }
+      moped_proj_components_subcomponents {
+        subcomponent_id
+      }
+      feature_street_segments {
+        id
+        geometry: geography
+        source_layer
+      }
+      feature_intersections {
+        id
+        geometry: geography
+        source_layer
+      }
+      feature_signals {
+        id
+        geometry: geography
+        source_layer
+      }
+      feature_drawn_lines {
+        id
+        geometry: geography
+        source_layer
+      }
+      feature_drawn_points {
+        id
+        geometry: geography
+        source_layer
+      }
```

Comment:
I see how the new `graphql-engine` relationships are created above linking `moped_proj_components` to these `feature_*` tables, which allows them to be drawn out here in the `moped_proj_components` portion of the GQL query. I also see the `project_geography` query, which feels like it might be pulling a very similar set of data. I think you have a comment below about using the view more, so there may be context I have not gotten to yet.